gst/nnstreamer/tensor_filter/tensor_filter.c

```diff
@@ -214,6 +219,9 @@ gst_tensor_filter_class_init (GstTensorFilterClass * klass)
   trans_class->transform_size =
       GST_DEBUG_FUNCPTR (gst_tensor_filter_transform_size);
 
+  /* setup sink event */
+  trans_class->sink_event = GST_DEBUG_FUNCPTR (gst_tensor_filter_sink_event);
```

Comment:
What's the rationale of using sink_event for "update model"?

Reply:
Well, in order to update model data with some way, I thought that a tensor filter needs to get a related event from somewhere. Actually, I have no idea yet how to send such events to a tensor filter. But, I think an easy way would be to use `gst_element_send_event()` for the filter if we implement gst app.

Anyway, AFAIK, `.send_event()` is called for pads such as sink/src depending on an event's direction. Actually, there was no rationale for why I chose `sink` instead of `src`.

Reply:
Would it be possible and easy for an app (an owner of pipeline) to invoke this event?

Reply:
Well, I'm not sure whether I understood your question correctly, but I'll describe my thought anyway. 

I think pipeline owners (i.e., gst app developers) can easily invoke this event because they can easily access the pointer of each element, and directly call `gst_element_send_event()`. In this case, on-device training may exist as a separated thread or process.

Another scenario is that on-device training co-exists in the same pipeline like the below figure.
![image](https://user-images.githubusercontent.com/48666767/70111612-24692f00-1696-11ea-8b68-ab708a042b44.png)
In this case, on-device training plugin needs to broadcast events to pipeline, and I'm not sure a tensor filter can receive events from a diverged branch like the below figure.
![image](https://user-images.githubusercontent.com/48666767/70111713-714d0580-1696-11ea-9faa-ffb51d99c2d8.png)

In fact, I have no clear working model yet, and we need to discuss more how model-updatable tensor filters and model-updater plugins (i.e., on-device training) can work together in nnstreamer.

Reply:
We need a small discussion with @jijoonmoon for this.